presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1201,6 +1203,22 @@ private PhysicalOperation visitScanFilterAndProject(
                 sourceLayout = source.getLayout();
             }
 
+            // filterExpression may contain large function calls; evaluate them before compiling.
```

Comment:
Curious: Is this part of optimization newly enabled by `RowExpression` ? :)

Reply:
Is was there in sqlToRowExpression translation opt mode.

Reply:
I will need to extract this to a common function. Ideally whenever `toRowExpression` was called originally, this common function should be called to optimized the expression.

Reply:
Update:  `LocalExecutionPlanner.toRowExpression` calls `SqlToRowExpressionTranslator.translate` with `optimize=true`: 

https://github.com/prestodb/presto/blob/2e199b0e64b346c6641b67b70ee442f1b0617c76/presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java#L1289-L1292

Which tries to use `ExpressionOptimizer` to optimize the generated `RowExpression`:

https://github.com/prestodb/presto/blob/2e199b0e64b346c6641b67b70ee442f1b0617c76/presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java#L168-L171

Looks like `ExpressionOptimizer` has some basic `RowExpression` evaluation ability to support basic optimizations such as constant folding (over `RowExpression`): https://github.com/prestodb/presto/blob/2e199b0e64b346c6641b67b70ee442f1b0617c76/presto-main/src/main/java/com/facebook/presto/sql/relational/optimizer/ExpressionOptimizer.java 

This is now done by the new introduced `RowExpressionInterpreter` :)

Reply:
Fixed in #12702